### PR TITLE
Certification test for Azure Blob storage state store component

### PR DESCRIPTION
**Pull request by tanvigour**
Signed-off-by: tanvigour <tanvi.gour@gmail.com>

# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1678

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by tanvigour** [comment]
@berndverst : I have qualified the non-aad tests locally for this, please review

**Comment by berndverst** [comment]
> @berndverst : I have qualified the non-aad tests locally for this, please review

Not sure what that means, but we should still include the `masterkey` authentication as a scenario / option here.

**Comment by tanvigour** [comment]
> > @berndverst : I have qualified the non-aad tests locally for this, please review
> 
> Not sure what that means, but we should still include the `masterkey` authentication as a scenario / option here.



> > @berndverst : I have qualified the non-aad tests locally for this, please review
> 
> Not sure what that means, but we should still include the `masterkey` authentication as a scenario / option here.

i meant, locally i have verified non-aad test qualification, but these changes do have aad authentication scenario as well.

**Review by berndverst — CHANGES_REQUESTED** [review]
Please add access key authentication. See binding certification test:
https://github.com/dapr/components-contrib/tree/master/tests/certification/bindings/azure/blobstorage

EDIT: Never mind, misunderstood your comment.

**Review by berndverst — CHANGES_REQUESTED** [review]
Please rebase this PR against the latest master branch and run `make modtidy-all`

File: tests/certification/state/azure/blobstorage/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Azure Blob storage certification testing
+
+This project aims to test the Azure Blob storage State Store component under various conditions.
+
+## Test plan
+
+## Basic Test for existing container for CRUD operations:
+1. Able to create and test connection.
+2. Able to do set, fetch, delete.
```
Comment (berndverst) [inline tests/certification/state/azure/blobstorage/README.md:9]: We really do not need to verify these things as this is all handled by conformance test. This just leads to code duplication.

All that needs to be tested is that we can authenticate and do anything.

**Review by berndverst — APPROVED** [review]

**Review by berndverst — APPROVED** [review]

**Pull request merged**